jpype/_jstring.py

```diff
@@ -62,22 +52,12 @@ def __getitem__(self, i):
             raise IndexError("Array index exceeds length")
         return self.charAt(i)
 
-    def __lt__(self, other):
-        return self.compareTo(other) < 0
-
-    def __le__(self, other):
-        return self.compareTo(other) <= 0
-
-    def __gt__(self, other):
-        return self.compareTo(other) > 0
-
-    def __ge__(self, other):
-        return self.compareTo(other) >= 0
-
     def __contains__(self, other):
         return self.contains(other)
 
     def __hash__(self):
+        if self == None: # lgtm [py/test-equals-none]
```

Comment:
it complains, because a None check in Python should be written as "is None" (for whatever reason).

Reply:
Yes they have it tagged as a performance warning.    True the None, True, and False are singletons, but that doesn't mean "is" is always the right answer.   I didn't feel like making a special function just for checking for null.